contracts/src/utils/math/alloy.rs

```diff
@@ -0,0 +1,167 @@
+//! Standard math utilities missing in `alloy_primitives`.
+use alloy_primitives::{uint, U256};
+
+/// Trait for standard math utilities missing in `alloy_primitives`.
+pub trait Math {
+    /// Returns the square root of a number. If the number is not a perfect
+    /// square, the value is rounded towards zero.
+    /// This method is based on Newton's method for computing square roots; the
+    /// algorithm is restricted to only using integer operations.
+    ///
+    /// # Arguments
+    ///
+    /// * `self` - value to perform square root operation onto.
+    #[must_use]
+    fn sqrt(self) -> Self;
+
+    /// Returns the average of two numbers. The result is rounded towards zero.
+    ///
+    /// # Arguments
+    ///
+    /// * `self` - first value to compute average.
+    /// * `rhs` - second value to compute average.
+    #[must_use]
+    fn average(self, rhs: Self) -> Self;
+}
+
+impl Math for U256 {
+    fn sqrt(self) -> Self {
```

Comment:
I think this one can be refactored, but may be later.
Square root implementation in [solidity](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/442886ed5ff8a0b9ab477b191f5238541ee6d772/contracts/utils/math/Math.sol#L407).